.github/shared/src/swagger.js

```diff
@@ -177,3 +261,68 @@ function json(file) {
   // Extension "json" with any case is a valid JSON file
   return typeof file === "string" && file.toLowerCase().endsWith(".json");
 }
+
+/**
+ * Get base name for swagger file
+ * @param {string} filePath - Path to swagger file
+ * @param {string} [version] - Version string to use for base name
+ * @returns {string} - Base name for the swagger file
+ */
+function getBaseNameForSwagger(filePath, version = "") {
+  if (version) {
+    const segments = filePath.split("/");
+    const index = segments.findIndex((v) => v.startsWith(version));
+    if (index !== -1) {
+      return segments.slice(index + 1).join("/");
+    }
+  }
+  return basename(filePath);
+}
+
+/**
+ * Extract version string from input file path
+ * @param {string} filePath - Path to the input file
+ * @param {boolean} [withPreview=false] - Whether to include preview suffix
+ * @returns {string} - Version string extracted from path, or empty string if not found
+ */
+export function getVersionFromInputFile(filePath, withPreview = false) {
```

Comment:
Inline this code in "get Version()".  You can still test it, by creating "new Swagger" from a test path, then calling "get Version()".